src/test/java/org/assertj/core/api/ThrowableAssertAlternative_havingCause_Test.java

```diff
@@ -0,0 +1,30 @@
+package org.assertj.core.api;
+
+import org.assertj.core.util.AssertionsUtil;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@DisplayName("ThrowableAssertAlternative havingCause")
+class ThrowableAssertAlternative_havingCause_Test {
+
+  @Test
+  void should_return_cause_if_throwable_has_cause() {
+    Throwable throwable = new Throwable("top level message", new Throwable("cause message"));
+    new ThrowableAssertAlternative<>(throwable)
+      .havingCause()
+      .withMessage("cause message");
+  }
+
+  @Test
+  void should_fail_if_throwable_has_no_cause() {
+    //WHEN
+    ThrowableAssert.ThrowingCallable code = () -> {
+      Throwable throwable = new Throwable("top level message");
+      new ThrowableAssertAlternative<>(throwable)
+        .havingCause();
+    };
+    //THEN
+    AssertionsUtil.assertThatAssertionErrorIsThrownBy(code).withMessage("expecting java.lang.Throwable: top level message to have a cause but it did not");
```

Comment:
this should build the error message with `shouldHaveCause`, here's an example from an `OptionalAssert` test
```java
  @Test
  public void should_fail_if_optional_is_empty() {
    // GIVEN
    String expectedValue = "something";
    // WHEN
    AssertionError error = expectAssertionError(() -> assertThat(Optional.empty()).contains(expectedValue));
    // THEN
    assertThat(error).hasMessage(shouldContain(expectedValue).create());
  }
```